src/codegen/solana_accounts/account_management.rs

```diff
@@ -0,0 +1,186 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::codegen::cfg::{ControlFlowGraph, Instr};
+use crate::codegen::{Builtin, Expression};
+use crate::sema::ast::{ArrayLength, Function, Namespace, StructType, Type};
+use crate::sema::solana_accounts::{DATA_ACCOUNT, SYSTEM_ACCOUNT};
+use num_bigint::BigInt;
+use num_traits::Zero;
+use solang_parser::pt::Loc;
+use std::collections::{HashSet, VecDeque};
+
+/// This function walks over the CFG and automates the account management, so developers do not need
+/// to do so. For instance, when calling 'new construct{address: addr}()', we construct the correct
+/// AccountMeta array with all the accounts the constructor needs.
+pub(crate) fn manage_contract_accounts(contract_no: usize, ns: &mut Namespace) {
+    let contract_functions = ns.contracts[contract_no].functions.clone();
```

Comment:
Why is this clone needed? Can't this be an iterator over `ns.contracts[contract_no].function`?

Reply:
I mutably borrow `ns.functions[func_no].cfg` in the loop. I cannot borrow `ns` twice.

Reply:
Ok I see. No need to bikeshed this